manifests/project.pp

```diff
@@ -82,12 +86,17 @@
   $create_project_opts = join(flatten([$dev_opt, "--prefer-${prefer}"]), ' ')
   $install_opts = join(flatten([$dev_opt, $script_opt, $custom_inst_opt, "--prefer-${prefer}" ]), ' ')
   $update_opts = join(flatten([$dev_opt, $script_opt, $custom_inst_opt, "--prefer-${prefer}", $lock_opt ]), ' ')
+  $home_directory = $user ? {
```

Comment:
That'd cause a lot of headaches since a composer command executed as non-root must have write access to `$target`, otherwise an error occurs. I'd suggest adding a `$composer_home` option instead which sets `COMPOSER_HOME`. But that's something for a different PR.